### ISSUE TITLE
Python profile

### ISSUE THREAD

**Issue by neworderofjamie** [project-rig/pynn_spinnaker#3]
Figure out where time is being spent in Python-side of P.R. attempt to optimise.

**Comment by neworderofjamie** [comment]
Has a very brief look at this - 90ish percent of time is spent in machine_controller i.e. IO bound so probably very low priority for now

**Comment by mundya** [comment]
Have you been thinking about how to load the machine in parallel? Windowing, if and when it's ever available, will be at most a factor 4 speed-up (I think I see an increase from 8Mib/s to 32Mib/s).

**Comment by neworderofjamie** [comment]
There's no reason not to load the machine in parallel. You'd just partition
`self.pop_synapse_vertices` and `self.pop_neuron_vertices` by their placements
proximity to ethernet ports hereish https://github.com/project-rig/pinus_rigida/blob/cpp/simulator.py#L394 (note that file is in desperate need of breaking up :)). I'd also like to make the loads in this loop
https://github.com/project-rig/pinus_rigida/blob/cpp/spinnaker/regions/synaptic_matrix.py#L96-L116
asynchronous so the latency can be at least partly hidden behind the numpy

**Comment by neworderofjamie** [comment]
Also, semi-amusingly, UDP is trashed by whatever lies between the two IT302 floor ports my Spinn2 and my desktop are connected to - switching to the board directly connected to my machine speeds it up by an order of magnitude

**Comment by mundya** [comment]
> whatever lies between the two IT302 floor ports my Spinn2 and my desktop are connected to

You'd rather not know, I'm sure.
